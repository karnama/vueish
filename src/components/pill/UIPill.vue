<template>
    <span :class="classes" class="rounded px-2">
        <slot>
            {{ label }}
        </slot>
    </span>
</template>

<script lang="ts">
import { computed, defineComponent } from 'vue';
import { label } from '@composables/input';
import { category } from '@composables/style';
import type { StyleType } from '@public-types';

const categoryClasses = {
    default: 'text-gray-700 bg-gray-200',
    primary: 'text-blue-700 bg-blue-200',
    info: 'text-blue-500 bg-blue-100',
    success: 'text-green-700 bg-green-200',
    warning: 'text-yellow-700 bg-yellow-200',
    danger: 'text-red-700 bg-red-200',
    brand: 'text-brand-700 bg-brand-200'
} as { [key in StyleType]: string; };

export default defineComponent({
    name: 'UIPill',

    props: {
        category,
        label
    },

    setup(props) {
        const classes = computed(() => categoryClasses[props.category as StyleType]);

        return { classes };
    }
});
</script>
