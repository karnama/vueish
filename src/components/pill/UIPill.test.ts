import { mount } from '@vue/test-utils';
import UIPill from './UIPill.vue';

describe('UIPill', () => {
    it('should display correctly', () => {
        const wrapper = mount(UIPill, {
            slots: {
                default: 'my-string'
            }
        });

        expect(wrapper.element).toMatchSnapshot();
    });
    it('should display the given label prop', () => {
        const wrapper = mount(UIPill, {
            props: {
                label: 'my-string'
            }
        });

        expect(wrapper.text()).toBe('my-string');
    });

    it('should display the default slot', () => {
        const wrapper = mount(UIPill, {
            slots: {
                default: 'my-string'
            }
        });

        expect(wrapper.text()).toBe('my-string');
    });
});
