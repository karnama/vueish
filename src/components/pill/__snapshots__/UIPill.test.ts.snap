// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIPill should display correctly 1`] = `
<span
  class="text-gray-700 bg-gray-200 rounded px-2"
>
  
  my-string
  
</span>
`;
